src/configurations.jl

```diff
@@ -89,22 +93,95 @@ function Configuration(orbs::Vector{Tuple{O,Int,Symbol}}) where {O<:AbstractOrbi
         push!(occupancy, occ)
         push!(states, state)
     end
-    Configuration(orbitals, occupancy, states)
+    Configuration(orbitals, occupancy, states, sorted=sorted)
 end
 
-Configuration(orbital::O, occupancy::Int, state::Symbol=:open) where {O<:AbstractOrbital} =
-    Configuration([orbital], [occupancy], [state])
+Configuration(orbital::O, occupancy::Int, state::Symbol=:open; sorted=false) where {O<:AbstractOrbital} =
+    Configuration([orbital], [occupancy], [state], sorted=sorted)
 
-Configuration{O}() where {O<:AbstractOrbital} =
-    Configuration(O[], Int[], Symbol[])
+Configuration{O}(;sorted=false) where {O<:AbstractOrbital} =
+    Configuration(O[], Int[], Symbol[], sorted=sorted)
 
 const RelativisticConfiguration{N} = Configuration{RelativisticOrbital{N}}
 
-issimilar(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital} =
-    a.orbitals == b.orbitals && a.occupancy == b.occupancy
+"""
+    issorted(cfg::Configuration)
+
+Tests if the orbitals of `cfg` is sorted.
+"""
+Base.issorted(cfg::Configuration) = cfg.sorted || issorted(cfg.orbitals)
+
+"""
+    sort(cfg::Configuration)
+
+Returns a sorted copy of `cfg`.
+"""
+Base.sort(cfg::Configuration) =
+    Configuration(cfg.orbitals, cfg.occupancy, cfg.states, sorted=true)
+
+"""
+    sorted(cfg::Configuration)
+
+Returns `cfg` if `cfg.sorted` or a sorted copy otherwise.
+"""
+sorted(cfg::Configuration) =
+    issorted(cfg) ? cfg : sort(cfg)
+
+"""
+    issimilar(a::Configuration, b::Configuration)
 
-Base.:(==)(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital} =
-    issimilar(a, b) && a.states == b.states
+Compares the electronic configurations `a` and `b`, only considering
+the constituent orbitals and their occupancy, but disregarding their
+ordering and states (`:open`, `:closed`, &c).
+
+# Examples
+
+```jldoctest
+julia> a = c"1s 2s"
+1s 2s
+
+julia> b = c"2si 1s"
+2sⁱ 1s
+
+julia> issimilar(a, b)
+true
+
+julia> a==b
+false
+```
+
+"""
+function issimilar(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital}
+    ca = sorted(a)
+    cb = sorted(b)
+    ca.orbitals == cb.orbitals && ca.occupancy == cb.occupancy
+end
+
+"""
+    ==(a::Configuration, b::Configuration)
+
+Tests if configurations `a` and `b`, considering orbital occupancy,
+ordering, and states.
+
+# Examples
+
+```jldoctest
+julia> c"1s 2s" == c"1s 2s"
+true
+
+julia> c"1s 2s" == c"1s 2si"
+false
+
+julia> c"1s 2s" == c"2s 1s"
+false
+```
+
+"""
+function Base.:(==)(a::Configuration{<:O}, b::Configuration{<:O}) where {O<:AbstractOrbital}
+    ca = sorted(a)
+    cb = sorted(b)
+    issimilar(ca, cb) && ca.states == cb.states && sortperm(a.orbitals) == sortperm(b.orbitals)
```

Comment:
Couldn't we just compare all the fields?

Reply:
b379f57